static/js/recent_topics_ui.js

```diff
@@ -707,6 +707,12 @@ function is_focus_at_last_table_row() {
     return row_focus === $topic_rows.length - 1;
 }
 
+function has_unread(row) {
+    const last_msg_id = topics_widget.get_current_list()[row].last_msg_id;
+    const last_msg = message_store.get(last_msg_id);
+    return last_msg.unread;
+}
```

Comment:
I'm not sure this algorithm is necessarily correct in general -- what you're checking is whether the last message in the topic is unread. We should be checking whether the topic has unread messages.

Do you want `unread.num_unread_for_topic(stream_id, topic)` instead?

Reply:
Looking at the patterns in this file, I think this is correct:

```
diff --git a/static/js/recent_topics_ui.js b/static/js/recent_topics_ui.js
index daadb38b76..a6cfe25901 100644
--- a/static/js/recent_topics_ui.js
+++ b/static/js/recent_topics_ui.js
@@ -710,7 +710,7 @@ function is_focus_at_last_table_row() {
 function has_unread(row) {
     const last_msg_id = topics_widget.get_current_list()[row].last_msg_id;
     const last_msg = message_store.get(last_msg_id);
-    return last_msg.unread;
+    return unread.num_unread_for_topic(last_msg.stream_id, last_msg.topic) > 0;
 }
 
 export function focus_clicked_element(topic_row_index, col, topic_key) {
```